### PR TITLE
Patient Dropdown Data Population and Refactor

### PR DESCRIPTION
**Pull request by cgolemme**
# Summary

This PR adds patient data from the bulk export server to the dropdown select box on the Patient Export card. Before, it was filled with fake data because the `/Patient` endpoint was not yet implemented on the bulk export server. 

Additionally, This PR refactors data fetching from `/Patient` and `/Group` endpoints of the bulk-data-server. This PR also refactors, the `export-type` component in order to follow best practices with the `useEffect` hook.

Also moved components associated with the query selector page into new folder inside components folder.

Note: if you run the server and the app and the app doesn't appear to make a call to the server it is because it is using the cached response from the last fetch. If this is causing an issue try `npm run build` or deleting the `fetch-cache` folder in the .next directory and running the app again. Make sure the bulk-export-server is running on port 3000 before running `npm run build`

## New behavior

Patient dropdown data is now populated from the bulk-export-server. With this change, I also moved the data fetching for both the patient dropdown and the group dropdown to the top level page component, and passed the fetched data down to the `export-type` component. This allowed me to use built-in `next.js` server side data fetching, which is preferred for something like populating a dropdown.

The ExportType component is now more streamlined. Instead of passing the react state value and setter function to its child components, `BuildQueryButton` and `Dropdown`, I moved the dropdown components to be in the ExportType component. This allows us to minimize the `useEffect` hook usage, which is best react practice. Also, now just the react state value is passed to BuildQuery button.

The BuildQueryButton component is now more succinct and readable (I think). I moved the conditions for the tooltip and disabled button to the top of the file for readability and combined the "system" and "patient"/"group" branches of this component as much as possible.

## Code changes

- `/src/app/layout.tsx` changed the metadata description to be a little better.
- `/src/app/page.module.css` removed some unused styles.
- `/src/app/query-selector/layout.tsx`:  removed the 'use client' in order to use async functions on the `page.tsx` file. Because of this change, I had to import Mantine components like the `Appshell.Header` directly (because of some strange mantine reason). Changed Text from `a` component to next `Link` component to follow best practices.
- `/src/app/query-builder/page.tsx`: added function to get data for dropdowns asynchronously and changed the page to render this data.
- `/src/app/query-builder/page.modules.css`: small style changes and removed unused ones.
- `/src/components/export-type.tsx` refactor and rename props.
-  `/src/components/build-query-button.tsx` refactor.
- `/src/components/dropdown.tsx` deleted and functionality moved into `export-type.tsx`.

# Testing guidance
- Start the bulk data server instructions [here](https://github.com/projecttacoma/bulk-export-server)
- `npm run check`
- `npm run dev`
- Go to http://localhost:3001/
- Navigate to query selector page
- Try selecting an id from the dropdown and click the button. Verify that the data on the next page is the selected data.
- Try selecting an id from both dropdowns and clicking the button on one of them. Make sure that the type of the export is correct.
- Verify that the button cannot be pressed if there is not a element selected, and that the tooltip appears.

### CODE REVIEW

File: src/components/query-selector/export-type.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import { useState } from 'react';
+import BuildQueryButton from './build-query-button';
+import { capitalize } from 'lodash';
+
+export type SupportedExportTypes = 'patient' | 'group' | 'system';
+
+export interface ExportTypeProps {
+  exportType: SupportedExportTypes;
+  dropdownData?: string[];
+}
```
Comment (elsaperelli) [inline src/components/query-selector/export-type.tsx:12]: At some point (not necessary right now) we may be using these more and may want to pull them out in a types file

File: src/components/query-selector/export-type.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const [queryId, setQueryId] = useState<string | null>('');
+
+  return (
+    <GridCol span={{ base: 12, md: 6, lg: 3 }} style={{ minWidth: rem(600) }}>
+      <Paper shadow="xl" radius="xl" p="xl">
+        <Stack>
+          <Center>
+            <Title>{`${capitalize(exportType)}-level Export`}</Title>
+          </Center>
+          {exportType == 'system' ? (
```
Comment (elsaperelli) [inline src/components/query-selector/export-type.tsx]: You can make this slightly more consider by using `??`! That allows you basically just to say "if" and have no "else". So you would do `{exportType == 'system' ?? (<Select ...`

File: src/components/query-selector/build-query-button.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  queryId: string | null;
+  exportType: SupportedExportTypes;
+}
+
+/*
+ *  Component for the build query button at the bottom of each export-type component
+ *  navigates to the next page, and creates the query string
+ */
+export default function BuildQueryButton({ queryId, exportType }: BuildQueryButtonProps) {
+  const toolTipDisabled = exportType === 'system' ? true : queryId ? true : false;
```
Comment (elsaperelli) [inline src/components/query-selector/build-query-button.tsx]: This certainly works but I can't help but wonder if there is some and/or logic that could make this slightly more readable LOL. You could do `toolTipEnabled`

File: src/components/query-selector/export-type.tsx (hunk 2 above)
Comment (elsaperelli) [reply src/components/query-selector/export-type.tsx]: Oh also `===` instead of `==`

File: src/components/query-selector/export-type.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+export interface ExportTypeProps {
+  exportType: SupportedExportTypes;
+  dropdownData?: string[];
+}
+
+/*
+ * Component for selecting which type of export the user wants.
+ */
+export default function ExportType({ exportType, dropdownData }: ExportTypeProps) {
+  const [queryId, setQueryId] = useState<string | null>('');
```
Comment (elsaperelli) [inline src/components/query-selector/export-type.tsx]: I think we either want to just make this type a string or initialize it as null? https://gist.github.com/mgramigna/bdc933fae2150216681c28d1f0f7aac7#hooks

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Just a few small comments but overall functionality looks great!

File: src/components/query-selector/build-query-button.tsx (hunk 3 above)
Comment (cgolemme) [reply src/components/query-selector/build-query-button.tsx]: Lol I kind of just extracted the operators I had in the components. I found a better way to do this.

File: src/components/query-selector/build-query-button.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  queryId: string | null;
+  exportType: SupportedExportTypes;
+}
+
+/*
+ *  Component for the build query button at the bottom of each export-type component
+ *  navigates to the next page, and creates the query string
+ */
+export default function BuildQueryButton({ queryId, exportType }: BuildQueryButtonProps) {
+  const buttonEnabled = exportType === 'system' || queryId != null;
```
Comment (elsaperelli) [inline src/components/query-selector/build-query-button.tsx]:
```suggestion
  const buttonEnabled = exportType === 'system' || queryId !== null;
```

**Review by elsaperelli — CHANGES_REQUESTED** [review]

**Review by elsaperelli — APPROVED** [review]
Lgtm!

**Pull request merged**